### PR TITLE
fix(azure): find formal images in community gallary

### PR DESCRIPTION
**Pull request by fruch**
since this is the way to lookup the images our use would use we are switching to this method, without it we can pickup unpromoated images, which can lead to rolling upgrades to fail, since base and target would be identical

### Testing

- [x] azure provision test

### PR pre-checks (self review)

- [x] I added the relevant `backport` labels
- [x] I didn't leave commented-out/debugging code

### Reminders

- Add New configuration option and document them (in `sdcm/sct_config.py`)
- Add unit tests to cover my changes (under `unit-test/` folder)
- Update the Readme/doc folder relevent to this change (if needed)

### CODE REVIEW

**Review by yaronkaikov — DISMISSED** [review]
LGTM

File: sdcm/provision/azure/utils.py [hunk 1]
```diff
@@ -93,11 +94,49 @@ def get_scylla_images(  # pylint: disable=too-many-branches,too-many-locals
...
+) -> list[CommunityGalleryImageVersion]:
+    branch_version = '.'.join(scylla_version.split('.')[:2])
+    if is_enterprise(branch_version):
+        gallery_image_name = f'scylla-enterprise-{branch_version}'
+    else:
+        gallery_image_name = f'scylla-{branch_version}'
+    community_gallery_images = azure_service.compute.community_gallery_image_versions.list(
+        location=region_name,
+        gallery_image_name=gallery_image_name,
+        public_gallery_name='scylladb-7e8d8a04-23db-487d-87ec-0e175c0615bb',
```
Comment (fruch) [inline sdcm/provision/azure/utils.py:112]: @yaronkaikov 

is there a place we can get this value ? (would it ever gonna change ?)

Comment (yaronkaikov) [reply sdcm/provision/azure/utils.py:112]: It is not going to change. we have it hard coded for promotion as well

**Comment by vponomaryov** [comment]
Adding backport for the `5.2` branch too because the upgrade test was recently added there: [scylla-5.2/rolling-upgrade/rolling-upgrade-azure-image-test#2](https://argus.scylladb.com/test/bb31edf1-38a5-42b3-9928-af2adf474c2e/runs?additionalRuns[]=bc2a5e45-52ac-4978-b46e-55e69353fcd5)

File: README.md [hunk 2]
```diff
@@ -83,6 +83,8 @@ hydra run-test longevity_test.LongevityTest.test_custom_time --backend azure --c
 
 # example of running specific docker version
 export SCT_SCYLLA_VERSION=5.2.1
+# we don't need reporting to argus for test local runs
+export SCT_USE_MGMT=false
```
[the remark this answers is not in the document]
Comment (fruch) [reply README.md:87]: the fact that it's part of this change,  it wasn't intended.

and yes, it should be argus that need to be disabled

**Comment by fruch** [comment]
> Adding backport for the `5.2` branch too because the upgrade test was recently added there: [scylla-5.2/rolling-upgrade/rolling-upgrade-azure-image-test#2](https://argus.scylladb.com/test/bb31edf1-38a5-42b3-9928-af2adf474c2e/runs?additionalRuns%5B%5D=bc2a5e45-52ac-4978-b46e-55e69353fcd5)

backport might be a bit trick, we need to be careful, and regenerate the docker image for that branch in the process

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (fruch) [reply README.md]: removed this change

**Comment by fruch** [comment]
> LGTM if it works

failed once on spot termination (meaning it was able to provision)
running now, once it would pass, I'll merge it.

I'll leave you the 5.2 backport, I'll do the rest of them

**Review by soyacz — APPROVED** [review]
LGTM

**Comment by fruch** [comment]
> > > LGTM if it works
> > > I'll leave you the 5.2 backport, I'll do the rest of them
> 
> @fruch Tried out the backport here: https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/valerii/job/vp-rolling-upgrade-azure-image-test/4/consoleFull Got the `AssertionError: scylla-server version isn't changed` error the same as without this change. Probably something else is missing? Or parameters are wrong? I used the same as in here: [scylla-5.2/rolling-upgrade/rolling-upgrade-azure-image-test#2](https://argus.scylladb.com/test/bb31edf1-38a5-42b3-9928-af2adf474c2e/runs?additionalRuns%5B%5D=bc2a5e45-52ac-4978-b46e-55e69353fcd5)

maybe the version you are trying is already promoted ?

**Comment by fruch** [comment]
> > > > LGTM if it works
> > > > I'll leave you the 5.2 backport, I'll do the rest of them
> > 
> > 
> > @fruch Tried out the backport here: https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/valerii/job/vp-rolling-upgrade-azure-image-test/4/consoleFull Got the `AssertionError: scylla-server version isn't changed` error the same as without this change. Probably something else is missing? Or parameters are wrong? I used the same as in here: [scylla-5.2/rolling-upgrade/rolling-upgrade-azure-image-test#2](https://argus.scylladb.com/test/bb31edf1-38a5-42b3-9928-af2adf474c2e/runs?additionalRuns%5B%5D=bc2a5e45-52ac-4978-b46e-55e69353fcd5)
> 
> maybe the version you are trying is already promoted ?

5.2.15 was used as base, and you are using the exact same version as target.
you can run it with `base_versions=5.2.15`

once a version is promoted, you can't rerun a rolling upgrade test with the same versions anymore.
this fix was for before it was promoted

**Comment by vponomaryov** [comment]
> > > > LGTM if it works
> > > > I'll leave you the 5.2 backport, I'll do the rest of them
> > 
> > 
> > @fruch Tried out the backport here: https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/valerii/job/vp-rolling-upgrade-azure-image-test/4/consoleFull Got the `AssertionError: scylla-server version isn't changed` error the same as without this change. Probably something else is missing? Or parameters are wrong? I used the same as in here: [scylla-5.2/rolling-upgrade/rolling-upgrade-azure-image-test#2](https://argus.scylladb.com/test/bb31edf1-38a5-42b3-9928-af2adf474c2e/runs?additionalRuns%5B%5D=bc2a5e45-52ac-4978-b46e-55e69353fcd5)
> 
> maybe the version you are trying is already promoted ?

Yes, you are right. I didn't check it today.

> once a version is promoted, you can't rerun a rolling upgrade test with the same versions anymore. this fix was for before it was promoted

As I understood the problem is general.
So, when `5.2.16-rc` build appears then we will be able to cover this case again, right?
And exactly for such cases this fix is needed. Right?

**Comment by vponomaryov** [comment]
Backport for the branch-5.2 is here: https://github.com/scylladb/scylla-cluster-tests/pull/7211

**Pull request merged**